### PR TITLE
Fix symlinking on osx-arm64

### LINKED ISSUE #19: python.app for osx-arm64 has a copy of python
- [x] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

When I create an environment with the `python.app` package for `osx-arm64`, the version of Python in the `python.app` package does not match the version in the environment. For example, if I create a new environment with `conda create -n b python.app python=3.8`, I get this behavior
```
[bkpoon@eeyore:~] conda activate b
(b) [bkpoon@eeyore:~] $CONDA_PREFIX/bin/python --version
Python 3.8.8
(b) [bkpoon@eeyore:~] $CONDA_PREFIX/python.app/Contents/MacOS/python --version
Python 3.8.6
```
From extracting the version 1.3 tarballs of the `osx-64` and `osx-arm64` packages from https://anaconda.org/conda-forge/python.app/files, it looks like the `osx-arm64` packages have a copy of python in `pythonapp/Contents/MacOS/python`

```
[bkpoon@eeyore:Downloads/a] ls -l pythonapp/Contents/MacOS 
total 8024
8024 -rwxr-xr-x@ 1 bkpoon  staff   3.9M Jan  8 03:36 python
```

 instead of a symbolic link.

```
[bkpoon@eeyore:Downloads/b] ls -l pythonapp/Contents/MacOS
total 0
0 lrwxr-xr-x@ 1 bkpoon  staff    19B Jan  8 03:54 python -> ../../../bin/python
```

This may also be the reason why I needed to copy the python for cross-compiling in Apple Silicon. The relevant pull request commit is https://github.com/conda-forge/cross-python-feedstock/commit/1c5b14a6f8c2d1c934ec6b3037423273a99aa0eb. I suspect that the copy would not be necessary if the symbolic link is restored.

I am not sure why the recipe is copying instead of making the symbolic link in `build.sh`, though.

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list
# packages in environment at /Users/bkpoon/software/miniforge3/envs/b:
#
# Name                    Version                   Build  Channel
ca-certificates           2020.12.5            h4653dfc_0    conda-forge
certifi                   2020.12.5        py38h10201cd_1    conda-forge
libcxx                    11.1.0               h168391b_0    conda-forge
libffi                    3.3                  h9f76cd9_2    conda-forge
ncurses                   6.2                  h9aa5885_4    conda-forge
openssl                   1.1.1k               h27ca646_0    conda-forge
pip                       21.0.1             pyhd8ed1ab_0    conda-forge
python                    3.8.8           h12cc5a1_0_cpython    conda-forge
python.app                1.3              py38he168f11_4    conda-forge
python_abi                3.8                      1_cp38    conda-forge
readline                  8.1                  hedafd6a_0    conda-forge
setuptools                49.6.0           py38h10201cd_3    conda-forge
sqlite                    3.35.4               hc49ca36_0    conda-forge
tk                        8.6.10               hf7e6567_1    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
xz                        5.2.5                h642e427_1    conda-forge
zlib                      1.2.11            h31e879b_1009    conda-forge

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info
    active environment : b
    active env location : /Users/bkpoon/software/miniforge3/envs/b
            shell level : 1
       user config file : /Users/bkpoon/.condarc
 populated config files : /Users/bkpoon/software/miniforge3/.condarc
                          /Users/bkpoon/.condarc
          conda version : 4.10.1
    conda-build version : not installed
         python version : 3.9.2.final.0
       virtual packages : __osx=11.2.3=0
                          __unix=0=0
                          __archspec=1=arm64
       base environment : /Users/bkpoon/software/miniforge3  (writable)
      conda av data dir : /Users/bkpoon/software/miniforge3/etc/conda
  conda av metadata url : https://repo.anaconda.com/pkgs/main
           channel URLs : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/osx-arm64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/osx-arm64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /Users/bkpoon/software/miniforge3/pkgs
                          /Users/bkpoon/.conda/pkgs
       envs directories : /Users/bkpoon/software/miniforge3/envs
                          /Users/bkpoon/.conda/envs
               platform : osx-arm64
             user-agent : conda/4.10.1 requests/2.25.1 CPython/3.9.2 Darwin/20.3.0 OSX/11.2.3
                UID:GID : 501:20
             netrc file : None
           offline mode : False

```
</details>

### LINKED ISSUE #20: @conda-forge-admin, rerender
- [ ] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

<br/>
Environment (<code>conda list</code>):
<details>

```
$ conda list

```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
$ conda info

```
</details>

### PR DESCRIPTION
**Pull request by conda-forge-linter**
Hi! This is the friendly automated conda-forge-webservice.

I've rerendered the recipe as instructed in #20.

Here's a checklist to do before merging.
- [ ] Bump the build number if needed.

Fixes #20
Fixes #19

### CODE REVIEW

**Comment by isuruf** [comment]
ping @conda-forge/python-app

**Comment by bkpoon** [comment]
Are there any lingering issues before this can be merged? Thanks!

**Comment by isuruf** [comment]
Nope. waiting for @conda-forge/python.app. @conda-forge/core, I'm adding myself as a maintainer. Merge?

File: .ci_support/osx_arm64_python3.9.____cpython.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ c_compiler:
 c_compiler_version:
 - '11'
 channel_sources:
-- conda-forge
+- conda-forge/label/rust_dev,conda-forge
```
Comment (dopplershift) [inline .ci_support/osx_arm64_python3.9.____cpython.yaml:8]: Including `rust_dev` here seems...odd?

Comment (xhochy) [reply .ci_support/osx_arm64_python3.9.____cpython.yaml:8]: We add that for all `osx-arm64` builds as currently only the rust nightlies are working there.

Comment (jakirkham) [reply .ci_support/osx_arm64_python3.9.____cpython.yaml:8]: That's done by the re-render ( https://github.com/conda-forge/python.app-feedstock/pull/21/commits/4f7bb8d92b7f9a28fee1196a635de271f21d8c08 ) and [comes from `conda-forge-pinning`]( https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/7af3a80bf666ea2b6af95096404b237f63554b52/recipe/conda_build_config.yaml#L105 ). I think this has to do with some issues building Rust on macOS ARM, but haven't followed that work closely

Comment (dopplershift) [reply .ci_support/osx_arm64_python3.9.____cpython.yaml:8]: Ah, now it makes sense. Thanks for the clarification.

**Pull request merged**